### ISSUE TITLE
ImportError: cannot import name 'cast' from 'typing_extensions' 

### ISSUE THREAD

**Issue by CrystalWindSnake** [zauberzeug/nicegui#1973]
### Description

When the version of the typing extensions library is below 4.7, the following error will be reported.

```
    from nicegui import ui,app,run
  File "C:\Users\PR\AppData\Local\Programs\Python\Python310\lib\site-packages\nicegui\__init__.py", line 1, in <module>
    from . import context, elements, run, ui
  File "C:\Users\PR\AppData\Local\Programs\Python\Python310\lib\site-packages\nicegui\ui.py", line 106, in <module>
    from .elements.badge import Badge as badge
  File "C:\Users\PR\AppData\Local\Programs\Python\Python310\lib\site-packages\nicegui\elements\badge.py", line 4, in <module>
    from .mixins.text_element import TextElement
  File "C:\Users\PR\AppData\Local\Programs\Python\Python310\lib\site-packages\nicegui\elements\mixins\text_element.py", line 3, in <module>
    from typing_extensions import Self, cast
ImportError: cannot import name 'cast' from 'typing_extensions' (C:\Users\PR\AppData\Local\Programs\Python\Python310\lib\site-packages\typing_extensions.py)
```

because `cast` was added to `typing_extensions` in version 4.7. In addition, cast should have been directly supported in Python 3.8.

[typing-extensions cast](https://typing-extensions.readthedocs.io/en/latest/index.html#cast)

**Comment by falkoschindler** [comment]
Thanks for spotting this, @CrystalWindSnake!

I updated several files to import `cast` from `typing` instead of `typing_extensions`.
And I updated the requirement to `typing_extensions>=4.0.0` because we use `Self` which was introduced in 4.0.0.

But actually our FastAPI requirement (>=0.93) induced `typing_extensions>=4.8.0`. Therefore your issue should already be fixed by cleanly re-installing NiceGUI with its dependencies. 🤔

**Comment by CrystalWindSnake** [comment]
Thank you for your quick response

**Comment by WolfgangFahl** [comment]
﻿I had to
```bash
pip install -U typing-extensions
```

pip install -U nicegui didn't help

**Comment by johnthagen** [comment]
For us, `tensorflow==2.13.0` caused this issue, checking it out with `poetry show typing_extensions --tree --why`

```
tensorflow-macos 2.13.0 TensorFlow is an open source machine learning framework for everyone.
└── typing-extensions >=3.6.6,<4.6.0
```

> But actually our FastAPI requirement (>=0.93)

We had a newer FastAPI version than that, but `typing_extensions` still resolved to 4.5.0.

To fix it, we updated to `tensorflow` 2.14.0, which removed the upper bound on `typing_extensions` and everything was fixed. But it does seem that there is some kind of issue here with `nicegui` 1.4.2.

**Comment by falkoschindler** [comment]
Oh, I was wrong: FastAPI 0.104.x requires `typing_extensions>=4.8.0`, but FastAPI 0.93.0 doesn't. This explain why NiceGUI's dependencies were indeed wrong in 1.4.2. But it should be fixed in 1.4.3 where we import `cast` from the built-in `typing` module.
